Common/src/gRPC/Services/IAgent.cs

```diff
@@ -39,49 +46,44 @@ public interface IAgent : IDisposable
   /// </returns>
   Task FinalizeTaskCreation(CancellationToken cancellationToken);
 
-  /// <summary>
-  ///   Process requests for creating child tasks
-  /// </summary>
-  /// <param name="requestStream">Collection of requests that represents the child tasks</param>
-  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
-  /// <returns>
-  ///   Reply sent to the worker describing the status of the child task creation
-  /// </returns>
-  Task<CreateTaskReply> CreateTask(IAsyncStreamReader<CreateTaskRequest> requestStream,
-                                   CancellationToken                     cancellationToken);
-
   /// <summary>
   ///   Get Common data from data storage as file in shared folder
   /// </summary>
-  /// <param name="request">Request specifying the data to retrieve</param>
+  /// <param name="resultId">Result id to retrieve</param>
+  /// <param name="token">Worker token for request validation</param>
   /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
   /// <returns>
-  ///   Response to send to the worker
+  ///   Id of the result sent to the worker
   /// </returns>
-  Task<DataResponse> GetCommonData(DataRequest       request,
-                                   CancellationToken cancellationToken);
+  Task<string> GetCommonData(string            resultId,
+                             string            token,
```

Comment:
nitpick: I would take the token first. Maybe you could also take multiple result Ids for the interface.